usr/bin/pia-manager

```diff
@@ -1,5 +1,54 @@
 #!/usr/bin/python3
 
-import os, getpass
+import os, sys, getpass
 
+import json
+import urllib.request
+
+
+def update_gateways(use_ips):
+    """Updates the list of PIA gateways. If `use_ips` is true, store IP addresses rather than hostnames."""
+    # TODO: also update the CRL from this call, can be handled in similar way most likely.
+    # grab new gateway json
+    response = urllib.request.urlopen('https://privateinternetaccess.com/vpninfo/servers?version=24')
+    data = response.read()
+    text = data.decode('utf-8')
+
+    # split json from CRL blob
+    server_info_text, crl = text.split('\n\n')
+    server_info = json.loads(server_info_text)
+
+    # assemble file that the manager uses
+    gateway_info = {}
+    for key, info in server_info.items():
+        if key == 'info':
+            continue
+
+        host = info['dns']
+        if use_ips:
+            host = info['openvpn_udp']['best'].split(':')[0]
+        gateway_info[key] = '{host} {name}'.format(host=host, name=info['name'])
+
+    # arrange by the region list we prefer
+    gateway_list = []
+    for key in server_info['info']['auto_regions']:
+        gateway_list.append(gateway_info[key])
+
+    gateways = '\n'.join(gateway_list)
+
+    # write out file
+    with open('/usr/share/pia-manager/gateways.list.dynamic', 'w') as fp:
```

Comment:
This would need root permissions I guess. You can do something like:

```python
if os.geteuid() != 0:
  exit ("Please run this program as root")
```

Before this point. Else it might fail with an exception.